File: the-lab-monorepo/apps/driverappios/driverappios/Utilities/SHA256Helper.swift
//
//  SHA256Helper.swift
//  driverappios
//

import CryptoKit
import Foundation

/// Returns the SHA-256 hex digest of a UTF-8 string.
func sha256Hex(_ input: String) -> String {
    let digest = SHA256.hash(data: Data(input.utf8))
    return digest.map { String(format: "%02x", $0) }.joined()
}

File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/util/Sha256Helper.kt
package com.pegasusx.driver.util

import java.security.MessageDigest

/** Returns the SHA-256 hex digest of a UTF-8 string (mirrors iOS sha256Hex). */
fun sha256Hex(input: String): String {
    val digest = MessageDigest.getInstance("SHA-256").digest(input.toByteArray(Charsets.UTF_8))
    return digest.joinToString("") { byte -> "%02x".format(byte) }
}
